src/context/ast/QueryAstContext.h

```diff
@@ -96,17 +96,20 @@ struct ReturnClauseContext final : CypherClauseContextBase {
     const YieldColumns*                          yieldColumns{nullptr};
     std::unique_ptr<OrderByClauseContext>        order;
     std::unique_ptr<PaginationContext>           pagination;
-    std::unordered_map<std::string, AliasType>*  aliases{nullptr};
+    std::unordered_map<std::string, AliasType>*  aliasesPtr{nullptr};
     // TODO: grouping columns
 };
 
 struct WithClauseContext final : CypherClauseContextBase {
     WithClauseContext() : CypherClauseContextBase(CypherClauseKind::kWith) {}
 
+    bool                                        distinct{false};
     const YieldColumns*                         yieldColumns{nullptr};
     std::unique_ptr<OrderByClauseContext>       order;
     std::unique_ptr<PaginationContext>          pagination;
     std::unique_ptr<WhereClauseContext>         where;
+    std::unordered_map<std::string, AliasType>* aliasesPtr{nullptr};
+    std::unordered_map<std::string, AliasType>  aliases;
```

Comment:
Why we need both value and ptr?

Reply:
aliasesPtr points to the previous clause's aliases. It's used when checking if a label used in the current clause is legal.

Reply:
If so, please given a meaningful name

Reply:
Maybe we can call it `previousAliases`

Reply:
> Why we need both value and ptr?

`value` means aliases that the current clause produces, and `ptr` means aliases that the current clause may use(it's produced by last clause)

Reply:
> Maybe we can call it `previousAliases`

aliasesUsed, aliasesGenerated